common/src/disk.rs

```diff
@@ -107,6 +107,7 @@ impl DatasetName {
         &self.pool_name
     }
 
+    // TODO: Maybe rename this to "kind"?
```

Comment:
👍 I figure this will have ripple effects which is why you didn't do it in this PR.

Reply:
Yeah, it's a mechanical change, so I figured I'd punt it out when I get to a less conflict-y place